### ISSUE TITLE
Getting Faraday::ConnectionFailed (execution expired) in the callback phase in Rails 5

### ISSUE THREAD

**Issue by Benratelade** [zquestz/omniauth-google-oauth2#284]
I am getting an error from Faraday after authorising the app in my browser. 

I found a [very similar thread on Stack Overflow](https://stackoverflow.com/questions/42711539/ror-google-oauth2-callback-phase-initiated-then-timeout) but the solution did not seem to work for me. 

My middleware config: 
```
Rails.application.config.middleware.use OmniAuth::Builder do
  provider :google_oauth2, ENV['GOOGLE_OAUTH_ID'], ENV['GOOGLE_OAUTH_SECRET', 
  {
  	scope: 'email,profile'
  }
end
```
I have tested a number of configurations here, with email only, profile only, various other services, plus.me etc...The Contacts and Google + APIs are definitely enabled in the developer console. 

The stack trace: 
```
I, [2017-06-05T23:17:02.133011 #21447]  INFO -- omniauth: (google_oauth2) Callback phase initiated.

Faraday::ConnectionFailed (execution expired):

/Users/benratelade/.rbenv/versions/2.3.2/lib/ruby/2.3.0/net/http.rb:880:in `initialize'
/Users/benratelade/.rbenv/versions/2.3.2/lib/ruby/2.3.0/net/http.rb:880:in `open'
/Users/benratelade/.rbenv/versions/2.3.2/lib/ruby/2.3.0/net/http.rb:880:in `block in connect'
/Users/benratelade/.rbenv/versions/2.3.2/lib/ruby/2.3.0/timeout.rb:101:in `timeout'
/Users/benratelade/.rbenv/versions/2.3.2/lib/ruby/2.3.0/net/http.rb:878:in `connect'
/Users/benratelade/.rbenv/versions/2.3.2/lib/ruby/2.3.0/net/http.rb:863:in `do_start'
/Users/benratelade/.rbenv/versions/2.3.2/lib/ruby/2.3.0/net/http.rb:852:in `start'
/Users/benratelade/.rbenv/versions/2.3.2/lib/ruby/2.3.0/net/http.rb:1398:in `request'
faraday (0.11.0) lib/faraday/adapter/net_http.rb:80:in `perform_request'
faraday (0.11.0) lib/faraday/adapter/net_http.rb:38:in `block in call'
faraday (0.11.0) lib/faraday/adapter/net_http.rb:85:in `with_net_http_connection'
faraday (0.11.0) lib/faraday/adapter/net_http.rb:33:in `call'
faraday (0.11.0) lib/faraday/request/url_encoded.rb:15:in `call'
faraday (0.11.0) lib/faraday/rack_builder.rb:139:in `build_response'
faraday (0.11.0) lib/faraday/connection.rb:377:in `run_request'
oauth2 (1.3.1) lib/oauth2/client.rb:99:in `request'
oauth2 (1.3.1) lib/oauth2/client.rb:146:in `get_token'
oauth2 (1.3.1) lib/oauth2/strategy/auth_code.rb:30:in `get_token'
omniauth-google-oauth2 (0.5.0) lib/omniauth/strategies/google_oauth2.rb:120:in `get_access_token'
omniauth-google-oauth2 (0.5.0) lib/omniauth/strategies/google_oauth2.rb:94:in `custom_build_access_token'
omniauth-oauth2 (1.4.0) lib/omniauth/strategies/oauth2.rb:73:in `callback_phase'
omniauth (1.6.1) lib/omniauth/strategy.rb:230:in `callback_call'
omniauth (1.6.1) lib/omniauth/strategy.rb:187:in `call!'
omniauth (1.6.1) lib/omniauth/strategy.rb:167:in `call'
omniauth (1.6.1) lib/omniauth/builder.rb:63:in `call'
rack (2.0.3) lib/rack/session/abstract/id.rb:232:in `context'
rack (2.0.3) lib/rack/session/abstract/id.rb:226:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/cookies.rb:613:in `call'
rack (2.0.3) lib/rack/etag.rb:25:in `call'
rack (2.0.3) lib/rack/conditional_get.rb:25:in `call'
rack (2.0.3) lib/rack/head.rb:12:in `call'
rack (2.0.3) lib/rack/session/abstract/id.rb:232:in `context'
rack (2.0.3) lib/rack/session/abstract/id.rb:226:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/cookies.rb:613:in `call'
activerecord (5.1.1) lib/active_record/migration.rb:556:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/callbacks.rb:26:in `block in call'
activesupport (5.1.1) lib/active_support/callbacks.rb:97:in `run_callbacks'
actionpack (5.1.1) lib/action_dispatch/middleware/callbacks.rb:24:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/executor.rb:12:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/debug_exceptions.rb:59:in `call'
web-console (3.5.1) lib/web_console/middleware.rb:135:in `call_app'
web-console (3.5.1) lib/web_console/middleware.rb:28:in `block in call'
web-console (3.5.1) lib/web_console/middleware.rb:18:in `catch'
web-console (3.5.1) lib/web_console/middleware.rb:18:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/show_exceptions.rb:31:in `call'
railties (5.1.1) lib/rails/rack/logger.rb:36:in `call_app'
railties (5.1.1) lib/rails/rack/logger.rb:24:in `block in call'
activesupport (5.1.1) lib/active_support/tagged_logging.rb:69:in `block in tagged'
activesupport (5.1.1) lib/active_support/tagged_logging.rb:26:in `tagged'
activesupport (5.1.1) lib/active_support/tagged_logging.rb:69:in `tagged'
railties (5.1.1) lib/rails/rack/logger.rb:24:in `call'
sprockets-rails (3.2.0) lib/sprockets/rails/quiet_assets.rb:13:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/remote_ip.rb:79:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/request_id.rb:25:in `call'
rack (2.0.3) lib/rack/method_override.rb:22:in `call'
rack (2.0.3) lib/rack/runtime.rb:22:in `call'
activesupport (5.1.1) lib/active_support/cache/strategy/local_cache_middleware.rb:27:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/executor.rb:12:in `call'
actionpack (5.1.1) lib/action_dispatch/middleware/static.rb:125:in `call'
rack (2.0.3) lib/rack/sendfile.rb:111:in `call'
railties (5.1.1) lib/rails/engine.rb:522:in `call'
puma (3.8.2) lib/puma/configuration.rb:224:in `call'
puma (3.8.2) lib/puma/server.rb:600:in `handle_request'
puma (3.8.2) lib/puma/server.rb:435:in `process_client'
puma (3.8.2) lib/puma/server.rb:299:in `block in run'
puma (3.8.2) lib/puma/thread_pool.rb:120:in `block in spawn_thread'
```
Any idea what might be causing this?

**Comment by zquestz** [comment]
First... the middleware config looks off. Should be:

```
Rails.application.config.middleware.use OmniAuth::Builder do
  provider :google_oauth2, ENV['GOOGLE_OAUTH_ID'], ENV['GOOGLE_OAUTH_SECRET'], 
  {
  	scope: 'email,profile'
  }
end
```

Then it merely makes basic requests to google for more information. No idea why that would fail, as these are simple HTTP gets.

**Comment by Benratelade** [comment]
Hi zquest and thank you for your quick reply. 

I tried again today, without changing anything in either my code or the developer console. And it works like a charm. 

Looking at the Stack Overflow issue I referred to in my question, and how that seemed to have started suddenly, and then started working after a minor change to scope, I will venture the guess that this probably has to do with Google taking a while to entirely enable the API. I'll see if I can replicate and might update your readme if I find anything. 

Sorry for the trouble
